BASE/src/ServerTelemetryChannel/Implementation/TransmissionPolicy/AuthenticationTransmissionPolicy.cs

```diff
@@ -0,0 +1,120 @@
+    /// </summary>
+    /// <remarks>
+    /// This class is disabled by default and expected to be enabled only when AAD has been configured in the AI SDK.
+    /// </remarks>
+    internal class AuthenticationTransmissionPolicy : TransmissionPolicy, IDisposable
+    {
+        private BackoffLogicManager backoffLogicManager;
+        private TaskTimerInternal pauseTimer = new TaskTimerInternal { Delay = TimeSpan.FromMinutes(1) };
+
+        public bool Enabled { get; set; } = false;
+
+        public override void Initialize(Transmitter transmitter)
+        {
+            if (transmitter == null)
+            {
+                throw new ArgumentNullException(nameof(transmitter));
+            }
+
+            this.backoffLogicManager = transmitter.BackoffLogicManager;
+
+            base.Initialize(transmitter);
+            transmitter.TransmissionSent += this.HandleTransmissionSentEvent;
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// This method subscribes to the <see cref="Transmitter.TransmissionSent"/> event.
+        /// This encapsulates all <see cref="HttpWebResponseWrapper.StatusCode"/> related to Authentication (AAD) scenarios.
+        /// </summary>
+        /// <remarks>
+        /// AN EXPLANATION OF THE STATUS CODES:
+        /// - <see cref="ResponseStatusCodes.BadRequest"/>
+        /// "HTTP/1.1 400 Incorrect API was used - v2 API does not support authentication".
+        /// This indicates that the AI resource was configured for AAD, but SDK was not.
+        /// This is a configuration issue and is not recoverable from the client side. 
+        /// If the customer chooses to disable AAD in the AI Resource, the SDK could resume sending.
+        /// - <see cref="ResponseStatusCodes.Unauthorized"/>
+        /// "HTTP/1.1 401 Unauthorized - please provide the valid authorization token".
+        /// This indicates that the authorization token was either absent, invalid, or expired.
+        /// The root cause is not known and we should throttle retries.
+        /// - <see cref="ResponseStatusCodes.Forbidden"/>
+        /// "HTTP/1.1 403 Forbidden - provided credentials do not grant the access to ingest the telemetry into the component".
+        /// This indicates the configured identity does not have permissions to publish to this resource.
+        /// This is a configuration issue and is not recoverable from the client side. 
+        /// This can be recovered if the user changes the AI Resource's configured Access Control.
+        /// </remarks>
+        private void HandleTransmissionSentEvent(object sender, TransmissionProcessedEventArgs e)
+        {
+            if (this.Enabled && e.Response != null)
+            {
+                switch (e.Response.StatusCode)
+                {
+                    case ResponseStatusCodes.BadRequest:
```

Comment:
could you confirm if BadRequest is not thrown by ingestion for anything else like a invalid json, or json exceeding size etc?

Reply:
I confirmed that none of our other Policies are handling 400.
I tested using invalid json and received;
`{"itemsReceived":1,"itemsAccepted":0,"errors":[{"index":0,"statusCode":400,"message":"Unknown domain type 'undefined'"}]}`

Reply:
Does this mean that, if we sent a bad json, ingestion responds with 400, and AuthenticationTransmissionPolicy would queue this item to be retried...400...retries..400...retries. (and forever, as the bad json will be forever a bad json)?

(The current behavior is - we drop the item if Ingestion responds with 400, and rightly so, as its pointless to retry a invalid json).

Reply:
AuthenticationTransmissionPolicy can ignore 400 as well and drop the data. If user disables AAD in portal, telemetry could be sent, but there is no way to distinguish this vs invalid jsob. So it might be better off not supporting retry for 400